packages/edit-site/src/components/sidebar-edit-mode/page-panels/swap-template-button.js

```diff
@@ -0,0 +1,151 @@
+/**
+ * WordPress dependencies
+ */
+import { useSelect, useDispatch } from '@wordpress/data';
+import { useMemo, useState, useCallback } from '@wordpress/element';
+import { decodeEntities } from '@wordpress/html-entities';
+import { __experimentalBlockPatternsList as BlockPatternsList } from '@wordpress/block-editor';
+import { Button, Modal } from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+import {
+	useEntityRecords,
+	useEntityRecord,
+	store as coreStore,
+} from '@wordpress/core-data';
+import { store as noticesStore } from '@wordpress/notices';
+import { parse } from '@wordpress/blocks';
+import { useAsyncList } from '@wordpress/compose';
+
+/**
+ * Internal dependencies
+ */
+import { store as editSiteStore } from '../../../store';
+
+export default function SwapTemplateButton() {
+	const [ showSwapTemplateModal, setShowSwapTemplateModal ] =
+		useState( false );
+	const { records: templates } = useEntityRecords(
+		'postType',
+		'wp_template',
+		{ per_page: -1 }
+	);
+	const currentTemplate = useSelect( ( select ) => {
+		const { getEditedPostType, getEditedPostId } = select( editSiteStore );
+		const { getEditedEntityRecord } = select( coreStore );
+		return getEditedEntityRecord(
+			'postType',
+			getEditedPostType(),
+			getEditedPostId()
+		);
+	}, [] );
+	const customTemplates = useMemo( () => {
+		return templates?.filter(
+			( template ) =>
+				template.is_custom &&
+				! template.has_theme_file &&
+				template.id !== currentTemplate.id &&
+				!! template.content.raw // Skip empty templates.
+		);
+	}, [ templates, currentTemplate.id ] );
+	const onClose = useCallback( () => setShowSwapTemplateModal( false ), [] );
+	if ( ! customTemplates?.length ) {
+		return null;
+	}
+	return (
+		<>
+			<Button
+				className="edit-site-page-panels__swap-template-button"
+				variant="secondary"
+				onClick={ () => setShowSwapTemplateModal( true ) }
+			>
+				{ __( 'Swap template' ) }
+			</Button>
+			{ showSwapTemplateModal && (
+				<Modal
+					title={ __( 'Swap template' ) }
+					onRequestClose={ onClose }
+					isFullScreen
+				>
+					<div className="edit-site-start-template-options__modal-content">
+						<TemplatesList
+							templates={ customTemplates }
+							currentTemplate={ currentTemplate }
+							onSelect={ onClose }
+						/>
+					</div>
+				</Modal>
+			) }
+		</>
+	);
+}
+
+function TemplatesList( { templates, currentTemplate, onSelect } ) {
+	const { postId, postType } = useSelect( ( select ) => {
+		const _context = select( editSiteStore ).getEditedPostContext();
```

Comment:
Question (unrelated): The selector looks to be deprecated (#46433) but is still used in a few places in the site editor. Should we un-deprecate it before there's a better alternative?

Reply:
Hm.. @youknowriad it seems even in your PR that deprecated, you still used it and there doesn't seem to be an alternative for that.. Any insights?